src/util/check.h

```diff
@@ -80,4 +86,25 @@ T&& inline_assertion_check(T&& val, [[maybe_unused]] const char* file, [[maybe_u
  */
 #define Assume(val) inline_assertion_check<false>(val, __FILE__, __LINE__, __func__, #val)
 
+/**
+ * NONFATAL_UNREACHABLE() is a macro that is used to mark unreachable code. It throws a NonFatalCheckError.
+ * This is used to mark code that is not yet implemented or is not yet reachable.
+ */
+#define NONFATAL_UNREACHABLE()                                                       \
+    do {                                                                             \
+        throw NonFatalCheckError(                                                    \
+            format_internal_error("Unreachable code reached",                        \
+                                  __FILE__, __LINE__, __func__, PACKAGE_BUGREPORT)); \
+    } while (0)
```

Comment:
```suggestion
    } while (false)
```

Reply:
`do-while-0` is only used when there is a need to force a trailing semicolon or to bundle several statements or "leaking" statements into one, none of which apply here, so it could be dropped as well?

Reply:
Concept ACK

Reply:
Fixed it